src/Query.cpp

```diff
@@ -84,12 +84,9 @@ void Query::add_apikey() {
 }
 
 pfc::string8 Query::url_encode(pfc::string8 in) {
-	in.replace_string("%", "%25", 0);	// escape % first, otherwise we'll be double escaping # or &
-	in.replace_string("#", "%23", 0);
-	in.replace_string("&", "%26", 0);
-	in.replace_string("+", "%2B", 0);
-
-	return in;
+	pfc::string8 encoded;
+	pfc::urlEncode(encoded, in);
```

Comment:
Interestingly, from my original testing, last.fm does not seem to want a real urlEncode to be called. From my testing, only these four characters should be encoded and the rest should be left as is. I can't explain how or why.

Reply:
OK. Maybe you should still copy the code for displaying output from the exception handler. That might be useful for future issues.

Reply:
Definitely. Will be giving him a custom build tonight with that code.